### PR TITLE
Use Route53 HealthCheck engine

### PR DESCRIPTION
**Pull request by AlexisMontagne**

### CODE REVIEW

**Comment by jlevesy** [comment]
LGTM

**Comment by jlevesy** [comment]
@AlexisMontagne  made some changes, RTM ?

File: dns-presence.go [hunk 1]
```diff
@@ -39,87 +51,118 @@ func init() {
...
-	)
+	log.Println("Applying changes")
 
-	<-sig
+	dnsConfiguration := &dns.Configuration{
+		Name:          flags.Names,
+		Type:          flags.Type,
+		TTL:           60,
+		Records:       []string{flags.Record},
+		SetIdentifier: flags.UnitName,
```
Comment (jlevesy) [inline dns-presence.go]: http://docs.aws.amazon.com/Route53/latest/APIReference/API_Change.html
UPSERT: If a resource record set does not already exist, Amazon Route 53 creates it. If a resource record set does exist, Amazon Route 53 updates it with the values in the request. Amazon Route 53 can update an existing resource record set only when all of the following values match: Name, Type, and SetIdentifier (for weighted, latency, geolocation, and failover resource record sets).

**Comment by jlevesy** [comment]
@AlexisMontagne What do you think of that ? We might need to test it on a sample dns unit first

File: dns/route53/manager.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+	"github.com/upfluence/dns-presence/dns"
+)
+
+type route53 struct {
+	client *goamz_route53.Route53
+}
+
+func NewRoute53DnsManager(client *goamz_route53.Route53) dns.Manager {
+	return &route53{
```
Comment (AlexisMontagne) [inline dns/route53/manager.go]: `return &route53{client}` is enough 😄

File: dns/route53/manager.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *route53) evaluateChanges(domains []string, config *dns.Configuration) (map[string][]goamz_route53.Change, error) {
+
```
Comment (AlexisMontagne) [inline dns/route53/manager.go]: trailing new line

File: dns/route53/manager.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	changes := map[string][]goamz_route53.Change{}
+
+	for _, zone := range zoneList.HostedZones {
+		for _, domain := range domains {
+			if strings.HasSuffix(domain, zone.Name) {
+				changes[zone.ID] = append(
+					changes[zone.ID],
+					goamz_route53.Change{
+						Action: "UPSERT",
+						Record: goamz_route53.ResourceRecordSet{
```
Comment (AlexisMontagne) [inline dns/route53/manager.go]: you should extract this struct instanciation in a separate method

File: healthcheck/route53/manager.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+func NewRoute53HealthcheckManager(client *goamz_route53.Route53) healthcheck.Manager {
+	return &route53{
+		client: client,
+	}
+}
+
+func (r *route53) FindHealthcheckIDByName(healthcheckName string) (string, error) {
+	healthcheckIDs, err := r.listHealtchecksIDs()
+	ID := ""
```
Comment (AlexisMontagne) [inline healthcheck/route53/manager.go]: `id` in lowercase would be more appropriate

Comment (AlexisMontagne) [inline healthcheck/route53/manager.go]: and may be declaring it as `var id string`

File: dns/route53/manager.go [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+	goamz_route53 "github.com/upfluence/goamz/route53"
+
+	"github.com/upfluence/dns-presence/dns"
+)
+
+type route53 struct {
+	client *goamz_route53.Route53
+}
+
+func NewRoute53DnsManager(client *goamz_route53.Route53) dns.Manager {
```
Comment (AlexisMontagne) [inline dns/route53/manager.go]: `DNS` is an acronym could you write it in full upper case 😄

File: dns/route53/manager.go (hunk 2 above)
Comment (jlevesy) [inline dns/route53/manager.go]: Make my linter shout

**Pull request merged**